qiskit/pulse/builder.py

```diff
@@ -773,39 +790,47 @@ def call_subroutine(
             self._compile_lazy_circuit()
             subroutine = self._compile_circuit(subroutine)
 
-        empty_subroutine = True
-        if isinstance(subroutine, Schedule):
-            if len(subroutine.instructions) > 0:
-                empty_subroutine = False
-        elif isinstance(subroutine, ScheduleBlock):
-            if len(subroutine.blocks) > 0:
-                empty_subroutine = False
-        else:
+        if not isinstance(subroutine, (Schedule, ScheduleBlock)):
             raise exceptions.PulseError(
                 f"Subroutine type {subroutine.__class__.__name__} is "
                 "not valid data format. Call QuantumCircuit, "
                 "Schedule, or ScheduleBlock."
             )
 
-        if not empty_subroutine:
-            param_value_map = {}
-            for param_name, assigned_value in kw_params.items():
```

Comment:
The change is because now `Call` instruction only takes `Schedule`. Since we have the reference mechanism for schedule block, we can create a reference instruction and assign the ``target`` immediately when schedule block is called as a target. (I lean toward deprecating Call in future, because of its complicated implementation which is hard to QPY serialize).

before:

- target is `Schedule` -> Wrap in `Call` instruction then append
- target is `ScheduleBlock` -> Wrap in `Call` instruction then append
- target is `QuantumCircuit` -> call schedule to create `Schedule` then wrap it in `Call` and append

now:

- target is `Schedule` -> same
- target is `QuantumCircuit` -> same
- target is `ScheduleBlock` -> create `Reference` and append, then call `assign_references` immediately with target
- target is `str` -> create `Reference`